### ISSUE TITLE
OSX Run as service - Agent never connects to VSTS

### ISSUE THREAD

**Issue by bc3tech** [Microsoft/vso-agent#283]
I've followed all the steps in the docs to configure and set up the agent, then move on to installing as a service.
When the agent is run interactively, it wires up to VSTS and shows "green" in the Agent pool list. However, when I Ctrl+C the interactive agent & follow the steps to install as service (`./agent/svc.sh install`), the service gets installed, status shows it's running, but the agent stays red in the VSTS agent pool and never picks up build requests.

If I'm missing something, it should probably be added to the docs. Otherwise, please let me know what I might be doing wrong.

Thanks

**Comment by bryanmacfarlane** [comment]
The docs refer to a svc status command.  What does that show?

**Comment by bc3tech** [comment]
```
Johns-MacBook-Pro-3:buildagent-internal3 brandonh$ ./agent/svc.sh status
action: status
serviceName: vsoagent.vsoagent.mine.Agent-Mac-Internal3

status vsoagent.mine.Agent-Mac-Internal3
17932	0	vsoagent.mine.Agent-Mac-Internal3

Success.
```

**Comment by bc3tech** [comment]
just tried installing another agent this morning using `curl` to get latest agent bits. Issue persists. works fine interactive, no dice as service.

**Comment by bryanmacfarlane** [comment]
The troubleshooting section shows where the trace/logs are at.

https://github.com/Microsoft/vso-agent/blob/master/docs/troubleshooting.md 

The most interesting one is likely at _diag folder since that's the listener.  Your status output shows return code of 0 and not running (-).

**Comment by bc3tech** [comment]
where do you see "not running (-)"?

I looked at the aforementioned troubleshooting doc and went to `/agent/` and there is no `_diag/` folder for me to look in, unfortunately.

any other ideas how we could troubleshoot this?

**Comment by bryanmacfarlane** [comment]
0.6.5 had a fix for running as a service.  Try that one.  If not, I can step you through the plist it generates on OSX

**Comment by bc3tech** [comment]
```
Enter Work Folder (press Enter for _work) > 
Enter Run Agent as service (press Enter for N) > Y
An error occured.  The method or operation is not implemented.
```

**Comment by bryanmacfarlane** [comment]
Log the issue in that repo.

Include details on Linux distribution and version
Include logs in _diag folder.

**Comment by bryanmacfarlane** [comment]
I still want to understand why this vsoagent isn't working for you

Like I said I fixed a bug in 0.6.5.  Try it.  All it does is create a plist file for you.  That's it.  The .services file in the agent root has the path to that plist file.  At that point it's the OS and troubleshooting OS X services.  Ensure the program and args in plist work from terminal

**Comment by bryanmacfarlane** [comment]
Try https://github.com/Microsoft/vsts-agent
